pymc/distributions/continuous.py

```diff
@@ -365,6 +365,10 @@ def __new__(cls, *args, **kwargs):
     @classmethod
     def dist(cls, *, size=None, **kwargs):
         res = super().dist([], size=size, **kwargs)
+        rng = res.owner.inputs[0]
+        # We set a dummy default_update that does not require sampling
+        if not hasattr(rng, "default_update"):
+            rng.default_update = rng
```

Comment:
I am not quite sure why this was not a problem in this test with the previous default_update. But if I do not include these lines, this test fails: https://github.com/pymc-devs/pymc/blob/5b31ec71389c56789578b110b111a08d230c712e/pymc/tests/test_sampling.py#L684

Reply:
I see now, updates are always executed, even if they don't involve variables in the graph, whereas default_updates are not (obviously)